### PR TITLE
add timeout to ActivateJobs request

### LINKED ISSUE #324: when using Camunda cloud zeebe, polling requests for activateJobs stuck forever
**Describe the bug**
when using Camunda cloud zeebe, polling requests for activateJobs stuck forever

**To Reproduce**
1. create camunda cloud channel 
2. register 1 task or more
3. wait a few days 
4. polling will stop working no new task reach to pyzeebe

**Expected behavior**
polling should work

### PR DESCRIPTION
**Pull request by EyalCMX**
add timeout to ActivateJobs request
handle ZeebeDeadlineExceeded exception

## Changes

- timeout to ActivateJobs request
- handle ZeebeDeadlineExceeded exception

### New Features *(required)*

None

### Deprecations *(required)*

None

### Enhancements *(optional)*

fixing the issue describe below:
when using Camunda cloud zeebe, polling requests for activateJobs can stuck forever,
which leads to inability to consume more tasks

## Checklist

- [ ] Unit tests
- [ ] Documentation

## References

Fixes #324

### CODE REVIEW

**Comment by JonatanMartens** [comment]
Hey @EyalCMX, looks good. 

Please rebase from master so that tests can run and we can merge

**Review by liorcmx — APPROVED** [review]
looks good

**Comment by EyalCMX** [comment]
> Hey @EyalCMX, looks good.
> 
> Please rebase from master so that tests can run and we can merge

@JonatanMartens done please approve!

**Comment by JonatanMartens** [comment]
@EyalCMX the job for format checking fails. 
Please format using black: `poetry run black .`, push the changes and I'll approve

**Comment by EyalCMX** [comment]
> @JonatanMartens done please approve!

@JonatanMartens done please approve!

**Comment by EyalCMX** [comment]
> @EyalCMX the job for format checking fails. Please format using black: `poetry run black .`, push the changes and I'll approve

@JonatanMartens done please approve!

**Comment by EyalCMX** [comment]
@JonatanMartens can u approve it again please

**Comment by EyalCMX** [comment]
@JonatanMartens  can u merge it i dont have permission

**Comment by michaelarnauts** [comment]
@EyalCMX @JonatanMartens i'm waiting for this to be merged. Is there a more maintained zeebe python plugin available?

**Comment by EyalCMX** [comment]
@michaelarnauts i dont think @JonatanMartens maintains this library... and i can't merge it

**Comment by xomiamoore** [comment]
Hey there @michaelarnauts @EyalCMX -- we're looking for additional maintainers for this project. https://github.com/camunda-community-hub/pyzeebe/issues/381

File: pyzeebe/grpc_internals/zeebe_job_adapter.py
Comment (JonatanMartens) [inline pyzeebe/grpc_internals/zeebe_job_adapter.py:39]: Would move 20 to a const named `DEFAULT_GRPC_REQUEST_TIMEOUT`.

**Comment by JonatanMartens** [comment]
Hey @EyalCMX, I'm very sorry for the delayed review

File: pyzeebe/grpc_internals/zeebe_job_adapter.py
Comment (EyalCMX) [reply pyzeebe/grpc_internals/zeebe_job_adapter.py]: done

File: pyzeebe/grpc_internals/zeebe_adapter_base.py
Comment (dimastbk) [inline pyzeebe/grpc_internals/zeebe_adapter_base.py:12]:
```suggestion
    ZeebeBackPressureError,
    ZeebeDeadlineExceeded,
    ZeebeGatewayUnavailableError,
    ZeebeInternalError,
```

File: pyzeebe/grpc_internals/zeebe_job_adapter.py
Comment (dimastbk) [inline pyzeebe/grpc_internals/zeebe_job_adapter.py:30]:
```suggestion
DEFAULT_GRPC_REQUEST_TIMEOUT = 20  # This constant represents the fallback timeout value

```

File: pyzeebe/worker/job_poller.py
Comment (dimastbk) [inline pyzeebe/worker/job_poller.py:10]:
```suggestion
    ZeebeBackPressureError,
    ZeebeDeadlineExceeded,
    ZeebeGatewayUnavailableError,
    ZeebeInternalError,
```

**Comment by liorcmx** [comment]
@EyalCMX , congrats!

**Pull request merged**
